documentation/src/test/java/example/DisplayNameGeneratorDemo.java

```diff
@@ -40,36 +39,16 @@ void if_it_is_negative(int year) {
 	}
 
 	@Nested
-	@DisplayNameGeneration(IndicativeSentences.class)
+	@DisplayNameGeneration(DisplayNameGenerator.IndicativeSentencesGenerator.class)
 	class A_year_is_a_leap_year {
 
 		@Test
 		void if_it_is_divisible_by_4_but_not_by_100() {
 		}
 
-		@ParameterizedTest(name = "Year {0} is a leap year.")
-		@ValueSource(ints = { 2016, 2020, 2048 })
-		void if_it_is_one_of_the_following_years(int year) {
-		}
-
-	}
-
-	static class IndicativeSentences extends DisplayNameGenerator.ReplaceUnderscores {
-
-		@Override
-		public String generateDisplayNameForClass(Class<?> testClass) {
-			return super.generateDisplayNameForClass(testClass);
-		}
-
-		@Override
-		public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
-			return super.generateDisplayNameForNestedClass(nestedClass) + "...";
-		}
-
-		@Override
-		public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
-			String name = testClass.getSimpleName() + ' ' + testMethod.getName();
-			return name.replace('_', ' ') + '.';
+		@DisplayName("if it is one of the following years.")
```

Comment:
How about the following?

```java
		@ParameterizedTest(name = "{0}")
		@ValueSource(ints = { 2016, 2020, 2048 })
		void if_it_is(int year) {
		}
```

Reply:
Interesting, I did not test the output with ParametrizedTest I'll check how it works and add this on the test's file

Reply:
The result of your example gives this output:
<img width="750" alt="Screen Shot 2019-12-17 at 12 17 20 PM" src="https://user-images.githubusercontent.com/17280833/71023045-8a5aab00-20c7-11ea-8918-c9dbeb946fd5.png">

In fact to DisplayGenerator is getting the "(int)" at the last of the name this also happens in ReplaceUndescore, 
the concatenation with the function `if_it_is()` with the separator and his parent class is working,
and output seems odd, because the parameter `name = "{0}"` just print the numbers, if we change that with a string name like: `@ParameterizedTest(name = " Year {0} is a leap year.")` will output this:

<img width="692" alt="Screen Shot 2019-12-17 at 12 38 33 PM" src="https://user-images.githubusercontent.com/17280833/71024594-deb35a00-20ca-11ea-988b-c9aadfc2b3d9.png">

I think the problem is that the **ParametrizedTest** data printing cycle does not go through **DisplayGenerator**, maybe those results could be outside the scope of this extension proposal?

Reply:
Fair enough.